In [340]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sbn

pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('max_colwidth', None)

artists_data_path = "../data/artists.jsonl"
sessions_data_path = "../data/sessions.jsonl"
track_storage_data_path = "../data/track_storage.jsonl"
tracks_data_path = "../data/tracks.jsonl"
users_data_path = "../data/users.jsonl"

artists_data_frame = pd.read_json(artists_data_path, lines=True)
sessions_data_frame = pd.read_json(sessions_data_path, lines=True)
track_storage_data_frame = pd.read_json(track_storage_data_path, lines=True)
tracks_data_frame = pd.read_json(tracks_data_path, lines=True)
users_data_frame = pd.read_json(users_data_path, lines=True)

Pominęliśmy analizę danych o przechowywaniu utworów, ponieważ nie wnoszą one żadnych informacji do analizy w naszym temacie projektu.

# Analiza artystów

In [341]:
artists_data_frame.head(n=5)

id             name  \
0  7b6Ui7JVaBDEfZB9k6nHL0  The Local Train   
1                      -1    Vishal Mishra   
2  6AETRCQep85mUtIlX7BaFV     Color Humano   
3  3PN8odvj48fnALdAsCKG3g           Crucis   
4  5SAeZRhgmJhgq196BR1Mna      Dúo Salteño   

                                                                      genres  
0  [desi pop, hindi indie, indian indie, indian rock, new delhi indie, sufi]  
1                                               [desi pop, modern bollywood]  
2                                            [argentine rock, rock nacional]  
3                                  [italian progressive rock, rock nacional]  
4                                     [folclore salteno, folklore argentino]

Atrybuty:
* id - identyfikator artysty
* name - nazwa artysty
* genres - gatunki muzyczne, które tworzy

In [342]:
artists_data_frame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10657 entries, 0 to 10656
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   id      10657 non-null  object
 1   name    10657 non-null  object
 2   genres  10126 non-null  object
dtypes: object(3)
memory usage: 249.9+ KB


### Analiza id

In [343]:
# find artists with -1 id
artists_data_frame[artists_data_frame['id'] == -1]

id                                          name  \
1      -1                                 Vishal Mishra   
14     -1                                      Luz Mala   
25     -1                              Mono & Nikitaman   
61     -1                                      Rebelion   
99     -1                                           FSB   
107    -1                                       LouKass   
158    -1                                 Quatro por Um   
161    -1                                      Mc Sapao   
213    -1                                 Fred Pellerin   
229    -1                                          Gölä   
256    -1                                    Zalo Reyes   
257    -1                               La Pozze Latina   
260    -1                               Fiskales Ad-Hok   
270    -1                                   Dario Dario   
274    -1                                  José Capmany   
289    -1                                    Synkopy 61   
296    -1                                     Debustrol   
312    -1             Disney - Arielle die Meerjungfrau   
361    -1                        Fede Finn & Funny Boyz   
379    -1                                 Tommy Seebach   
399    -1                                Ramon Leonardo   
404    -1                                    Maria Diaz   
413    -1                               Sonia Silvestre   
441    -1                                  El Andariego   
448    -1                                 Kolumbus Kris   
471    -1                                   Luis Lucena   
472    -1                                        Haizea   
533    -1                                  Gordo Master   
537    -1                                  Las Carlotas   
539    -1                                      Els Pets   
549    -1                                 Musta Paraati   
653    -1                                Kotiteollisuus   
710    -1                               Viky Moscholiou   
714    -1                                  Panos Kiamos   
718    -1                             Miltos Pashalidis   
735    -1                               Christos Dantis   
746    -1                                  Fotis Siotas   
749    -1                             Anastasios Rammos   
752    -1                                   Yorgos Rous   
790    -1                                   Astro Bunny   
796    -1                                            鄭興   
826    -1                                     Ámokfutók   
850    -1                                       Dzsúdló   
907    -1                                     Superglad   
932    -1                      The Dublin City Ramblers   
942    -1                                           Ash   
955    -1                                    Mr. Bungle   
992    -1                                     Hayehudim   
1004   -1                                 Yardena Arazi   
1005   -1                                   Yoav Itzhak   
1006   -1                                 Gary Eckstein   
1022   -1                                  Lucille Crew   
1024   -1                                     Dan Toren   
1027   -1                                 Shlomi Shaban   
1065   -1                                Helgi Skúlason   
1124   -1                                Gunni og Felix   
1130   -1                                        Sveppi   
1132   -1                                        Grafík   
1155   -1                                   Jón Jónsson   
1163   -1                                         Attan   
1176   -1                          Modena City Ramblers   
1209   -1                                     Tricarico   
1231   -1                   Norihiko Hashida And Climax   
1233   -1                                          COCO   
1255   -1                                       はっぴいえんど   
1262   -1                                Stardust Revue   
1271   -1                                   Number Gir

Istnieje 557 artystów, których id jest ustawione na -1.
TODO co z tym zrobic?

In [344]:
print("unique ids: {}/{}".format(artists_data_frame['id'].nunique(), artists_data_frame['id'].count()))

unique ids: 10101/10657


Nieunikatowe ID wynikają z tego, że istnieje 557 artystów, których ID jest ustawione na -1.

### Analiza name


In [345]:
print("unique names: {}/{}".format(artists_data_frame['name'].nunique(), artists_data_frame['name'].count()))

# find artists with duplicated names
artists_data_frame[artists_data_frame['name'].duplicated(keep=False)]

unique names: 10643/10657


id               name  \
358    2xgbz6gqoIF4S6JZoRiH2y             Dalton   
390    38YroIcEOM2zsIEGdtmZYm              Bamse   
460    0Yk0PoyjQiEyIKgnaJFR0n            Traffic   
1093   3rUSk3kcpl4Nseu1DqFNpJ                 KK   
1384   3lPK4zqijDKAvZkdlmZRfN              Mango   
2076   4ppDL5W65K1HG6EF7yaDLd  Little River Band   
2352   53Thxvlr6imD5y3lxouOu4            Phoenix   
2546   363VbwpX9anvrThJ2qDwBy                TNT   
2721   1CqOLQmjzVWXQTiIN5Wucs                TNT   
2757   7c5278WShlxTk3vS9XoTf8              Wings   
3967                       -1               Eloy   
4314   1FizW0swe8ODtXvEOHXbHX                LFO   
5192   3Vszg64mczF1dWEcW2dU5W            Luciano   
6283   2jFXKAH9A9H8lVnJkkLxXl              Omega   
6495                       -1            Traffic   
6803   0eet2VBconGGtNYJ0CBgHu               Eloy   
6959   0UCa56b7hiX05ZWC5BH2zo             Nikita   
7561   1UjxAZqzphB1tsMb1aWBj0              Omega   
7887   1xU878Z1QtBldR7ru9owdU            Phoenix   
8398   3sFhA6G1N0gG1pszb6kk1m              Wings   
9113   1V95p8oWGxXlpcFRYqXV3w              Bamse   
9262   6clbbhnIqpHnqxwtOWcilg  Little River Band   
9286   3CJKkU0XuElRT1z8rEtIYg            Luciano   
9455   4fEkbug6kZzzJ8eYX6Kbbp                 KK   
9461   5iBphIrmXEjfrqiuz5y6pY              Mango   
9869   2M0T4a1pkOC5nifN9W6e9e                LFO   
9944   7tS2WuFuYfT8wcw0dy63D9             Nikita   
10467  25JJnYKQwlu4uPHJP0pN5l             Dalton   

                                                                                                                                                                                                                                  genres  
358                                                                                                                                                                                                                 [classic danish pop]  
390                                                                                                                                                                                                              [barnmusik, bornesange]  
460                                                                                                                                                                                                        [estonian pop, estonian rock]  
1093                                                                                                                                                                                              [classic icelandic pop, icelandic pop]  
1384                                                                                                                                                                                                                    [lithuanian pop]  
2076                                                                                                                                                         [album rock, classic rock, country rock, folk rock, mellow gold, soft rock]  
2352                                                                                                                                                                                                                     [romanian rock]  
2546                                                                                                                                                                                                             [glam metal, hard rock]  
2721                                                                                                                                                                                                     [euphoric hardstyle, hardstyle]  
2757                                                                                                                                                                    

Istnieje 28 artystów, których nazwa jest zduplikowana.
TODO co z tym zrobic?


### Analiza genres


In [346]:
# find artists with null genres
artists_data_frame[artists_data_frame['genres'].isnull()]

id  \
24     058bioyBUzSkKCfrqvODLu   
26     3N4ZBIXZfvF0VLgEclgJvX   
85     3fxtMTmbY0jgtjHDuXspus   
122    0cDZ1YAlkkIn0Z7BjiEUj1   
125    5CorrQPbOid8RKf5rCpxG7   
135    2u7kjqj92WlpCSAKiOeZ6I   
169    28rjo4eRvpmLW0EsCIy6FW   
187    4fk1ToCb0m7irMHCaZhIrM   
190    0hGCbqEUzKxBUuJ7Q8ZFvK   
208    3IYryWetzBVax8oJPkgXMk   
217    5x0wy0trcqYLvwUG5PFmQL   
231    7wkPBPwF9oOZJ8lEbQjIVt   
249    7C7z1OMQ7TOtFBkwzK0zX9   
254    0oGbFTtR8UqMpgDFLpFKc1   
357    4dj3Rt7naflzJgPcFkOJH5   
365    2Fgt9tP2vJusWCaJK3e9QR   
373    1SrmGTb6i2OfOmNkbX3Ft4   
380    44os6d6JylvrXEyQLYeqGr   
417    46aJHc5ahjcNZfiS44i1zo   
487    1r5SGuosYsz2ruhoBgJ4iF   
490    7nOYYLDwTAvcZ4Hy5tOry0   
537                        -1   
574    627dKDcsgfO25s2ScC4sgQ   
613    6JlgvkzzxFaMfDpu7zxUav   
614    7zD8bBoHnqnQsjYWEOe3V8   
646    3hMNlyrxz0vtDz70ewLHgl   
654    2ECoOJ3zAmnbZoCwTaf9A2   
676    5Jh4UFOCWwleD3kcd88xyh   
712    1crGTvNXrAffvbpXZyOCKx   
745    5HRyAD6yCLqZlzh4R4ds1O   
772    4C7zRUPI7U5RVXGbpgeg2b   
798    6pnkoq165sVlU8H3WZKnhn   
807    3eeg2cVDU1rRPrvIwa0FlX   
837    68yBeY4QowunDr8foW1VaI   
839    32E92dXwYIiCE5ORdXRVOD   
871    77X41f0zoHXGhC3a3UvdJk   
874    4RLzLs2SHeyA0nLLlAJBg4   
884    126aQ4k98tuE3BXgfPwHpX   
896    5sFfUahhUdAKlrr2AUyMx6   
929    5w2TwcoIzksdooTyczidj3   
941    6lyku6lhbB3tLc3Dghs5CM   
945    4zPGlyitRvJo7iz6OKlETa   
953    6a2HcrBBLrhSq57wzuoXYr   
964    1F102kNzMqsmOpF7AfFmm5   
976    5ETZ4zl1tAJe8zCTDEaoGO   
978    110wHDAsaXDaKKdUztwGlE   
991    3giniJpo19Md4V5Plataq2   
1018   4VxlxOXkJFI9QRdBHUDIC4   
1038   2Eij3bHbG52JBSEqXSwga3   
1046   6eRpJMvXvYgm6apZRfoQWk   
1053   2q1LRGJHpFxovU8Tz6OgRn   
1094   5IRON8eFfbctCsLo6HlSdQ   
1095   1rJkz5vopfGxTUGFNB3o4G   
1097   01w8eDOz2L9SFgv1ukZPRv   
1099   0my8yB6MM8YVbIR8A96GY3   
1161   3vvugAMkG44OwAjfds1QpR   
1165   5XcM6fOuamexiFlGtmMRwq   
1187   3uQsgKw2TZkZRbxAeEdZfW   
1258   2E3u3DFA1rwICiHqlXJ05L   
1352   5ctTSlA3JZKProh0uSzTa8   
1357   5fd9qkcJWbNiFtlGGaoVIr   
1432   4oopAFtLdKXCoq2XoEjDZD   
1437   1KiCGFfdGc5RDccoRXHFjw   
1456   7GIvsDVOV1bjghuisC9opx   
1469   5A9NWhs9ydYZPGgvcWI8Ms   
1471   2ow8nzGoMwOtNtiupXpANu   
1474   5sWCGfIxmNiMAl6n8kdbkb   
1494   1QRod4LR53Ouoq8XoGdF4Q   
1504   1VeH892FTsVyB6XAnMvEbL   
1508   22XVSTMZerjq8l2QxXNMFY   
1546   2T6hbG0qQTyZglPgH7UEMP   
1556   3FsNh5N2AxOGA1UU5IAnlG   
1592   0bYI0rlhxJWiOD2ZRKbb7w   
1602   1F30RWETubxDmbyfxeVPg7   
1626   0ob2mO2kOzEdaGLYSth531   
1663   4WLGcWrkSExCqILxDk7ol6   
1686   2lzeV1nxm8MB7z85L53JV2   
1709   5icZHFQjI6SRsvXNOKUXZ1   
1734   6Y99TxJKt19wJFbmpoRzLp   
1763   1QfTCSQTI4rrK027UaiSQ6   
1786   7s5xlRV8yGHwxkPggVKdN6   
1793   5gsErkGsLeCYCSK0Pxj01N   
1822   1c9LVBvYq58ViD8qdLml7A   
1838   1dmEUAQBopYUrsMnQnScWL   
1857   72eP0W3rIhkxd0NHGg4w4u   
1873   58BqVfV4jCaxcbdZnmUvxK   
1922   6ELAHXutcbgmM3v0Bg2nei   
1947   0Nf2qv7lBcOJ2u3tH0YPXX   
1998   0Utyuwr2dSRI7b7JDaBfbK   
2017   5pbzj8JGUpUCMHESd1yHFv   
2036   56CaueCWXmgGQ9sFaUTeMa   
2090   6wYJgFEDVZmiCywTxOmoEe   
2110   7n1EM42Fseebd9H2p8eWWS   
2114   1RSKoT6hnzaLjvL0LCF0ew   
2133   0FRKTwQSToXpCxYMhyUzYY   
2135   3O9bFJBTRddC4HOsZixhuw   
2179   28WXX79U4jzq1kfjx5QXPy   
2182   1nEmAvnOKMBYCziZQNb5B4   
2191   3kUC3a0wTkjsdkJiytULkN   
2202   2HVjYynnJqLIg4HVV2A31m   
2215   2JDzHpbTcm5cu8Ne2SDYo4   
2224   4XI9RntQXI5Ly7j87g9y9I   
2256   1ts2oIXKCpWSRFnR78bulp   
2277   4Scx8KcZlLA5ORsj5Gejci   
2280   5O9SEPgz10bQCeiIctR08i   
2319   5uOMOTzmfhliUjnyiJh0kn   
2356   6XuwlE6QNkhGKauGl4jM7t   
2357                       -1   
2367   5W6FFILqUpVfpsZC9KadWN   
2372   1Tm5STqbzJ39okZK1y6UWK   
2379   6l6cvAPpAXHxNHYzSHHTCu   
2404   0PqNs9XCK5D16ihtRMuOoU   
2419   7LMlaqzutxVEEEung8DPbE   
2434   2iHiqwlS40GTFt8eELeoHr   
2444   2OU0OH164ypvP4G3AYynXs   
2447   4DzKphc3l03DORh036Vhrw   
2458   0IyeWjNLB0Lxw0MgCEZtHw   
2474   2GqX0GnrKdi6Z2X2OReYOK   
2481   4w8YQxx0XQfWZgQwRSqcoE   
2487   7zYj9S9SdIunYCfSm7vzAR   
2507   1bufaOa1xsdvxGxeqQJtzW   
2

Istnieje 531 artystów, których gatunki nie zostały określone.
TODO co z tym zrobic?


# Analiza sesji

In [347]:
sessions_data_frame.head(n=5)

timestamp  user_id                track_id event_type  \
0 2023-01-04 11:46:25.000    101.0  78ybROgQ2V3TySb8ZfPNdg       PLAY   
1 2023-01-04 11:49:32.533      NaN  2lkoxPYi7o9OlaGP9bVI09       PLAY   
2 2023-01-06 04:38:08.000    101.0  0juIv1JPqoCrWku2nIfMiA       PLAY   
3 2023-01-06 04:40:49.544    101.0  0juIv1JPqoCrWku2nIfMiA       LIKE   
4 2023-01-06 04:44:14.267    101.0  4srmb7zgB7eKWLUL96xftd       PLAY   

   session_id  
0         124  
1         124  
2         125  
3         125  
4         125

Atrybuty:
* timestamp - czas rozpoczęcia sesji
* user_id - identyfikator użytkownika
* track_id - identyfikator utworu
* event_type - typ zdarzenia (PLAY, LIKE, SKIP)
* session_id - identyfikator sesji


In [348]:
sessions_data_frame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3971 entries, 0 to 3970
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   timestamp   3971 non-null   datetime64[ns]
 1   user_id     3776 non-null   float64       
 2   track_id    3776 non-null   object        
 3   event_type  3763 non-null   object        
 4   session_id  3971 non-null   int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(2)
memory usage: 155.2+ KB


### Analiza timestamp

In [349]:
sessions_data_frame['timestamp'].describe(datetime_is_numeric=True)

count                             3971
mean     2023-02-09 22:20:59.431924736
min                2023-01-01 01:12:53
25%      2023-01-22 12:22:54.168499968
50%      2023-02-08 12:59:38.910000128
75%      2023-02-28 06:16:43.153500160
max         2023-03-24 06:28:07.464000
Name: timestamp, dtype: object

Sesje rozpoczęły się 2023-01-01 01:12:53 i zakończyły 2023-03-24 06:28:08.464000.


### Analiza user_id

In [350]:
# average number of sessions per user
print("avg number of sessions per user {}".format(sessions_data_frame['user_id'].value_counts().mean()))

# min number of sessions per user
print("min number of sessions per user {}".format(sessions_data_frame['user_id'].value_counts().min()))

# max number of sessions per user
print("max number of sessions per user {}".format(sessions_data_frame['user_id'].value_counts().max()))


avg number of sessions per user 75.52
min number of sessions per user 20
max number of sessions per user 148


Średnia liczba sesji na użytkownika wynosi 75.5, a maksymalna 148.


### Analiza track_id



In [351]:
# TODO


### Analiza event_type

In [352]:
sessions_data_frame['event_type'].value_counts()


PLAY             2242
SKIP              825
LIKE              619
BUY_PREMIUM        47
ADVERTISEMENT      30
Name: event_type, dtype: int64

### Analiza session_id

In [353]:
# average number of events per session
print("avg number of events per session {}".format(sessions_data_frame['session_id'].value_counts().mean()))


avg number of events per session 6.2437106918239


Średnia liczba zdarzeń na sesję wynosi 6.2


# Analiza danych o utworach


In [354]:
tracks_data_frame.head(n=5)


id                                     name  \
0  4LdczeQKU4ojwFJd6QWGnW  Taking Care Of Business - 2005 Remaster   
1  7F20EOraDwU8KvNcyzUKVz                                     Leef   
2  5EMGRFIidh98v9eXj6QnOU                              Dime Que No   
3  5XFfcpc0V1fqHeiEjJd4y5             I Hate the Capitalist System   
4  4fsmo5mukKd4vafcHN41KP                       Hai la sârbă roată   

   popularity  duration_ms  explicit               id_artist release_date  \
0         1.0       177853         0  4AgFg0hHythyklEs4t7XKt   2011-01-01   
1        42.0       217903         0  5l2Xy4aUoJDRSpsYHyOumD   2015-06-26   
2         NaN       266013         0  0h1zs4CTlU9D2QtgPxptUD   2012-01-23   
3        28.0       196387         0  7htyUKqAPLpbTA9nkEhGRf   1973-01-01   
4        24.0       223013         0  4a0FNCbvDEoeHYFfc045W4   2008-01-01   

   danceability  energy  key  loudness  speechiness  acousticness  \
0         0.554   0.847    7    -7.237       0.0633       0.00789   
1         0.673   0.727    9    -6.587       0.0401       0.28200   
2         0.535   0.648    0    -5.842       0.0364       0.36900   
3         0.454   0.115    5   -15.617       0.0405       0.79700   
4         0.509   0.699   10    -4.935       0.0372       0.68600   

   instrumentalness  liveness  valence    tempo  
0          0.000002     0.183    0.733   90.655  
1          0.000000     0.178    0.782  121.016  
2          0.000000     0.075    0.311  142.486  
3          0.000000     0.103    0.366   86.761  
4          0.000016     0.267    0.949  174.827

Atrybuty:
* id - identyfikator utworu
* name - nazwa utworu
* popularity - popularność utworu
* duration_ms - długość utworu w milisekundach
* explicit - czy utwór zawiera treści nieodpowiednie dla dzieci
* id_artist - identyfikator artysty
* release_date - data wydania utworu
* danceability - współczynnik "taneczności" utworu
* energy - współczynnik "energii" utworu
* key - główny ton utworu
* loudness - głośność utworu
* speechiness - współczynnik "mowy" utworu
* acousticness - współczynnik "akustyczności" utworu
* instrumentalness - współczynnik "instrumentalności" utworu
* liveness - współczynnik "żywotności" utworu
* valence - współczynnik "pozytywności" utworu
* tempo - tempo utworu

In [355]:
tracks_data_frame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21608 entries, 0 to 21607
Data columns (total 17 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id                20496 non-null  object 
 1   name              20557 non-null  object 
 2   popularity        20505 non-null  float64
 3   duration_ms       21608 non-null  int64  
 4   explicit          21608 non-null  int64  
 5   id_artist         20526 non-null  object 
 6   release_date      21608 non-null  object 
 7   danceability      21608 non-null  float64
 8   energy            21608 non-null  float64
 9   key               21608 non-null  int64  
 10  loudness          21608 non-null  float64
 11  speechiness       21608 non-null  float64
 12  acousticness      21608 non-null  float64
 13  instrumentalness  21608 non-null  float64
 14  liveness          21608 non-null  float64
 15  valence           21608 non-null  float64
 16  tempo             21608 non-null  float6

### Analiza id
TODO

### Analiza name
TODO

### Analiza popularity
TODO

### Analiza duration_ms
TODO

### Analiza explicit
TODO

### Analiza id_artist
TODO

### Analiza release_date
TODO

### Analiza danceability
TODO

### Analiza energy
TODO

### Analiza key
TODO

### Analiza loudness
TODO

### Analiza speechiness
TODO

### Analiza acousticness
TODO

### Analiza instrumentalness
TODO

### Analiza liveness
TODO

### Analiza valence
TODO

### Analiza tempo
TODO

# Analiza użytkowników

In [356]:
users_data_frame.head(n=5)

user_id             name      city               street  \
0      101    Klara Herdzik  Szczecin   pl. Floriana 55/22   
1      102  Szymon Plebanek    Kraków   al. Kalinowa 27/40   
2      103     Szymon Zawal    Gdynia    al. Głogowa 14/10   
3      104  Andrzej Berendt    Kraków  ul. Strażacka 14/63   
4      105     Ida Karsznia    Kraków        al. Złota 549   

                      favourite_genres  premium_user   id  
0     [permanent wave, mandopop, funk]           1.0  NaN  
1                                 None           1.0 -1.0  
2      [filmi, regional mexican, folk]           1.0  NaN  
3          [tropical, new wave, latin]           1.0  NaN  
4  [soul, tropical, alternative metal]           1.0  NaN

Atrybuty:
* user_id - identyfikator użytkownika
* name - imię i nazwisko użytkownika
* city - miasto zamieszkania użytkownika
* street - ulica zamieszkania użytkownika
* favourite_genres - ulubione gatunki użytkownika
* premium_user - czy użytkownik posiada konto premium
* id - zbędny atrybut


In [357]:
users_data_frame.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   user_id           50 non-null     int64  
 1   name              50 non-null     object 
 2   city              50 non-null     object 
 3   street            50 non-null     object 
 4   favourite_genres  48 non-null     object 
 5   premium_user      49 non-null     float64
 6   id                5 non-null      float64
dtypes: float64(2), int64(1), object(4)
memory usage: 2.9+ KB


### Analiza user_id
TODO

### Analiza name
TODO

### Analiza city
TODO

### Analiza street
TODO

### Analiza favourite_genres
TODO

### Analiza premium_user
TODO

### Analiza id
TODO
